05_stringer/alextmz/main_test.go

```diff
@@ -0,0 +1,51 @@
+package main
+
+import (
+	"bytes"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+)
+
+func Test_IPAddrString(t *testing.T) {
+	var tests = map[string]struct {
+		arg  IPAddr
+		want string
+	}{
+		"empty IPv4": {
+			IPAddr{},
+			"0.0.0.0"},
+		"8bits IPv4": {
+			IPAddr{254},
+			"254.0.0.0"},
+		"16bits IPv4": {
+			IPAddr{254, 254},
+			"254.254.0.0"},
+		"24bits IPv4": {
+			IPAddr{254, 254, 254},
+			"254.254.254.0"},
+		"full IPv4": {
+			IPAddr{254, 254, 254, 254},
+			"254.254.254.254"},
+	}
+
+	for name, tt := range tests {
+		test := tt
```

Comment:
rather that using both `test` and `tt`, where `tt` is kind of invented because you need another name, you can just use the one name and shadow that inside the loop:
```
for name, test := range tests {
    test := test
    ...
}
```

When an experienced Go reviewer sees that, they understand exactly what is going on. Usually you would not want to shadow variables because it can cause confusion, but this is one of those cases where it is almost always done due to a wart in the language and makes sense.

Reply:
Agreed.